File: backend/app/schemas/workout.py
from datetime import datetime, date
from decimal import Decimal
from typing import List, Optional
from pydantic import BaseModel, ConfigDict


class ExerciseSearchResult(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    name: str
    category: str
    muscle_group:         Optional[str]   = None
    equipment:            Optional[str]   = None
    level:                Optional[str]   = None
    met_value:            float
    # Phase 6 — wger image + muscle data
    image_url_thumb:      Optional[str]   = None
    primary_muscle_ids:   Optional[str]   = None
    secondary_muscle_ids: Optional[str]   = None


class WorkoutLogCreate(BaseModel):
    exercise_id:  Optional[int]   = None
    log_date:     date
    sets:         Optional[int]   = None
    reps:         Optional[int]   = None
    weight_kg:    Optional[float] = None
    duration_min: Optional[float] = None
    notes:        Optional[str]   = None


class WorkoutLogRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    user_id: str
    log_date: date
    exercise_id: Optional[int] = None
    exercise_name: str
    category: str
    sets: Optional[int] = None
    reps: Optional[int] = None
    weight_kg: Optional[float] = None
    duration_min: Optional[float] = None
    calories_burned: Optional[float] = None
    notes: Optional[str] = None
    created_at: datetime
    # Phase 6 — denormalized from exercise_library at read time
    image_url_thumb:      Optional[str]   = None
    primary_muscle_ids:   Optional[str]   = None
    secondary_muscle_ids: Optional[str]   = None


class WorkoutLogUpdate(BaseModel):
    sets:         Optional[int]   = None
    reps:         Optional[int]   = None
    weight_kg:    Optional[float] = None
    duration_min: Optional[float] = None
    notes:        Optional[str]   = None


class DailyWorkoutRead(BaseModel):
    log_date: date
    entries: List[WorkoutLogRead]
    total_calories_burned: float

File: frontend/src/types/workout.ts
export interface Exercise {
  id: number;
  name: string;
  category: string;
  muscle_group:         string | null;
  equipment:            string | null;
  level:                string | null;
  met_value:            number;
  // Phase 6 — wger image + muscle data
  image_url_thumb:      string | null;
  primary_muscle_ids:   string | null;  // "4;2" semicolon-separated
  secondary_muscle_ids: string | null;
}

export interface WorkoutLogEntry {
  id: number;
  user_id: string;
  log_date: string;
  exercise_id: number | null;
  exercise_name: string;
  category: string;
  sets: number | null;
  reps: number | null;
  weight_kg: number | null;
  duration_min: number | null;
  calories_burned: number | null;
  notes: string | null;
  created_at: string;
  // Phase 6 — denormalized from exercise_library
  image_url_thumb:      string | null;
  primary_muscle_ids:   string | null;
  secondary_muscle_ids: string | null;
}

export interface DailyWorkout {
  log_date: string;
  entries: WorkoutLogEntry[];
  total_calories_burned: number;
}
